kafka-client/src/main/java/dev/responsive/kafka/internal/utils/SharedClients.java

```diff
@@ -1,32 +1,87 @@
 package dev.responsive.kafka.internal.utils;
 
+import dev.responsive.kafka.api.config.StorageBackend;
 import dev.responsive.kafka.internal.config.InternalConfigs;
 import dev.responsive.kafka.internal.db.CassandraClient;
+import dev.responsive.kafka.internal.db.mongo.ResponsiveMongoClient;
 import java.util.Map;
+import java.util.Optional;
 import org.apache.kafka.clients.admin.Admin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Basic container class for session clients and other shared resources that should only
  * be closed when the app itself is shutdown
  */
 public class SharedClients {
-  public final CassandraClient cassandraClient;
-  public final Admin admin;
+
+  private final StorageBackend storageBackend;
```

Comment:
I'd prefer not to store this as a separate field, but instead to return the backend type based on whether the mongo or C* client is present -- let's figure it out based on the actual clients so we're guaranteed to reflect reality